modules/core/src/classes/matrix4.ts

```diff
@@ -371,7 +389,9 @@ export class Matrix4 extends Matrix {
    * @param result
    * @returns self
    */
-  getTranslation(result: NumericArray = [-0, -0, -0]): NumericArray {
+  getTranslation(): Vector3;
+  getTranslation<T extends NumericArray>(result: T): T;
+  getTranslation(result = new Vector3()): Vector3 {
```

Comment:
Should we change the return type here in the implementation, as it is not always a Vector3?

Reply:
Well, I guess we can keep the NumericArray return type in the default case, that still allows to chain calls if a Vector3 is provided as input... I'll update that !

Reply:
Thanks. My understanding when using overloaded signatures is that the "implementation" signature doesn't become visible to the user. Only the overloaded signatures without implementation become part of the public API. 

If I am correct, the change I am proposing is more a matter of making the code a little clearer to maintain.

Reply:
Fixed